pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -670,7 +670,8 @@ func (c *Cluster) failoverMon(name string) error {
 		return errors.Wrap(err, "failed to place new mon on a node")
 	}
 
-	if c.spec.Network.IsHost() {
+	isHost, _ := c.spec.Network.IsHost()
```

Comment:
We shouldn't make it a habit to ignore error return values.
Ditto for other similar changes as well.

Reply:
@BlaineEXE  wrote:

> We shouldn't make it a habit to ignore error return values. Ditto for other similar changes as well.

ok, good point. Agreed. I will address this, but it might make the code flow more complicated and less natural.

Reply:
The complication is that `IsHost()` returns an error, but I don't think it should. The flow will be much simpler if `IsHost()` only returns a bool. There can be a single check in the reconcile for valid network settings, instead of every time `IsHost()` is called. This will also allow you to revert a lot of changes in this PR.

Reply:
@travisn  wrote: 
> The complication is that `IsHost()` returns an error, but I don't think it should.

The decision to have IsHost return error on invalid nrywork spec was a ewsult of the discussion between @BlaineEXE  and myself in the original PR #13878
https://github.com/rook/rook/pull/13878#discussion_r1513348424





>  The flow will be much simpler if `IsHost()` only returns a bool.
>
> There can be a single check in the reconcile for valid network settings, instead of every time `IsHost()` is called. 
Agreed.


> This will also allow you to revert a lot of changes in this PR.

This sounds like it is desirable to have as few changes as possible in one PR.
Of course, fewer changes tend to be simpler and therefore less likely to break things. So I do see the point. 😄

While it does seem kind of sad to me to revert a major piece of the work put into this so far, I will do it as I see the logic behind the reasoning.

Reply:
@travisn  starting to undo the changes that add the eroor return to IsHost(), I am wondering if it would make sense to still validate the network spec in IsHost() but just issue an informational log message and return false for host network instead of erroring out. wdyt?

Reply:
Since `IsHost()` is called from multiple locations, the log entry could easily be too verbose and repetitive. If there is an error condition to check for, I would recommend have a separate method that checks the error once during a reconcile, then `IsHost()` can assume there is no error condition and just return `bool`.